File: shipfinity.frontend/src/app/shared/models/product.ts
export default class Product {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  discountPercentage: number;
  rating: number;
  imageUrl?: string;

  constructor(
    id?: number,
    name?: string,
    description?: string,
    price?: number,
    categoryId?: number,
    discountPercentage?: number,
    rating?: number,
    imageUrl?: string
  ) {
    this.id = !id ? 0 : id;
    this.name = !name ? 'no name' : name;
    this.description = !description ? 'no description' : description;
    this.price = !price ? 0 : price;
    this.categoryId = !categoryId ? 0 : categoryId;
    this.discountPercentage = !discountPercentage ? 0 : discountPercentage;
    this.rating = !rating ? 0 : rating;
    this.imageUrl = !imageUrl ? undefined : imageUrl;
  }
}

export interface ProductEdit {
  id?: number | null;
  name: string;
  description: string;
  price: number;
  categoryId: number;
}

export interface ProductDetails {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  imageUrl?: string;
  rating: number;
  discountPercentage: number;
}

export interface ReviewProductDto {
  comment: string;
  rating: number;
}

File: Shipfinity.Domain/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shipfinity.Domain.Models
{
    public class Product : BaseEntity
    {
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;
        [MaxLength(255)]
        public string Description { get; set; } = string.Empty;
        public Category Category { get; set; }
        public int CategoryId { get; set; }
        public Seller Seller { get; set; }
        public int SellerId { get; set; }
        [Range(0, 999999)]
        public double Price { get; set; }
        [Range(0, 100)]
        public int DiscountPercenrage { get; set; }
        [Range(0, 5)]
        public short Rating { get; set; }
        [MaxLength(150)]
        public string? ImageUrl { get; set; }
        public List<ProductOrder> ProductOrders { get; set; } = new();
    }
}
